lib/sdf-server/src/service/public/components.rs

```diff
@@ -0,0 +1,186 @@
+
+    for (key, value) in payload.domain.into_iter() {
+        // Update the property
+        let prop_id = key.prop_id(&ctx, schema_variant.id).await?;
+        let prop = Prop::get_by_id(&ctx, prop_id).await?;
+        let attribute_value_id =
+            Component::attribute_value_for_prop_id(&ctx, component_id, prop_id).await?;
+        let av = AttributeValue::get_by_id(&ctx, attribute_value_id).await?;
+        let before_value = av.value(&ctx).await?;
+        AttributeValue::update(&ctx, attribute_value_id, Some(value.clone())).await?;
+
+        // Log the property update
+        ctx.write_audit_log(
+            AuditLogKind::UpdatePropertyEditorValue {
+                component_id,
+                component_name: component_name.clone(),
+                schema_variant_id,
+                schema_variant_display_name: schema_variant_display_name.clone(),
+                prop_id,
+                prop_name: prop.name.clone(),
+                attribute_value_id,
+                before_value,
+                after_value: Some(value),
+            },
+            prop.name.clone(),
+        )
+        .await?;
+        let parent_prop = match Prop::parent_prop_id_by_id(&ctx, prop_id).await? {
+            Some(parent_prop_id) => Some(Prop::get_by_id(&ctx, parent_prop_id).await?),
+            None => None,
+        };
+
+        // Send the property update event to posthog
+        tracker.track(
+            &ctx,
+            "property_value_updated",
+            serde_json::json!({
+                "how": "/public/component/property_value_updated",
+                "component_id": component_id,
+                "component_schema_name": schema.name(),
+                "prop_id": prop_id,
+                "prop_name": prop.name,
+                "parent_prop_id": parent_prop.as_ref().map(|prop| prop.id),
+                "parent_prop_name": parent_prop.as_ref().map(|prop| prop.name.clone()),
+                "change_set_id": ctx.change_set_id(),
+            }),
+        );
+    }
+
+    // Send the WsEvent indicating the component was updated
+    let mut socket_map = HashMap::new();
+    let payload = component
+        .into_frontend_type(
+            &ctx,
+            None,
+            component.change_status(&ctx).await?,
+            &mut socket_map,
+        )
+        .await?;
+    WsEvent::component_updated(&ctx, payload)
+        .await?
+        .publish_on_commit(&ctx)
+        .await?;
+
+    // Commit the changes
+    ctx.commit().await?;
+
+    Ok(Json(UpdateComponentPropertiesResponse {}))
+}
+
+#[derive(Deserialize)]
+struct UpdateComponentPropertiesPath {
+    component_id: ComponentId,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct UpdateComponentPropertiesRequest {
+    domain: HashMap<ComponentPropKey, serde_json::Value>,
+}
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct UpdateComponentPropertiesResponse {}
```

Comment:
Since these structs are not public, I have been naming them just Response/Request (and by this logic, Path), since this reduces boilerplate on usage and avoids weird scenarios were we copy controllers to create new ones and end up with names that make zero sense accidentaly. 

Whatever you choose is fine though, just a suggestion

Reply:
I love that convention! The reason I have them named is that I'm experimenting with the readability of having all change_set routes in one file (since they are often super related). That just requires either inline modules or names. I'll experiment with inline modules (I think there is a bunch of value in having the CRUD that manipulates a single object together in a single module) in the next PR!